warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -339,203 +346,72 @@ public boolean intersect(List<JexlNode> delayedNodes) {
+            merged.count = -1;
+            merged.uids = ImmutableSortedSet.of();
+            
         } else {
             
-            if (o.isInfinite() && isInfinite()) {
-                /*
-                 * D) Both sub trees are UNKNOWN, so we must propagate everything
-                 */
-                merged.count = -1;
-                merged.uids = ImmutableSortedSet.of();
-            } else {
-                if (onlyEvents()) {
-                    /*
-                     * E) We have small AND LARGE
-                     */
-                    merged.count = count;
-                    
-                    HashMultimap<String,JexlNode> ids = HashMultimap.create();
-                    for (IndexMatch match : uids) {
-                        JexlNode newNode = match.getNode();
-                        if (null != newNode)
-                            ids.put(match.uid, newNode);
-                    }
-                    
-                    JexlNodeSet ourDelayedNodes = new JexlNodeSet();
-                    ourDelayedNodes.addAll(delayedNodes);
-                    // we may actually have no node on o
-                    if (null != o.getNode())
-                        ourDelayedNodes.add(o.getNode());
-                    
-                    Set<IndexMatch> matches = buildNodeList(ids, IndexMatchType.AND, true, Lists.newArrayList(ourDelayedNodes.getNodes()));
-                    
-                    merged.uids = ImmutableSortedSet.copyOf(matches);
-                    merged.count = merged.uids.size();
-                } else if (o.onlyEvents()) {
-                    /*
-                     * E) We have LARGE AND SMALL
-                     */
-                    HashMultimap<String,JexlNode> ids = HashMultimap.create();
-                    for (IndexMatch match : o.uids) {
-                        JexlNode newNode = match.getNode();
-                        if (null != newNode)
-                            ids.put(match.uid, newNode);
-                    }
-                    
-                    JexlNodeSet ourDelayedNodes = new JexlNodeSet();
-                    ourDelayedNodes.addAll(delayedNodes);
-                    // possible, depending on how query is processed that we have no node.
-                    if (null != getNode())
-                        ourDelayedNodes.add(getNode());
-                    
-                    Set<IndexMatch> matches = buildNodeList(ids, IndexMatchType.AND, true, Lists.newArrayList(ourDelayedNodes.getNodes()));
-                    merged.uids = ImmutableSortedSet.copyOf(matches);
-                    merged.count = merged.uids.size();
-                } else {
-                    
-                    merged.count = Math.min(count, o.count);
-                    merged.uids = ImmutableSortedSet.of();
-                }
-            }
+            // else we are left countably infinite shard ranges. Take the minimum count.
```

Comment:
Here is where we should explain why we are choosing the shard instead of the UIDs.